### PR TITLE
Instance page

### PR DESCRIPTION
**Pull request by krawieck**
only info tab for now

Screenshots:
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 10
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 21
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 30
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 38
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 33

### CODE REVIEW

**Comment by krawieck** [comment]
Screenshots:
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 10
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 21
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 30
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 38
Simulator Screen Shot - iPhone 11 - 2020-09-08 at 23 11 33

File: lib/pages/communities_list.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    return Scaffold(
+        appBar: AppBar(
+          title: Text(title ?? '', style: theme.textTheme.headline6),
+          centerTitle: true,
+          backgroundColor: theme.cardColor,
+          iconTheme: theme.iconTheme,
+        ),
+        body: ListView.builder(
+          itemBuilder: (context, i) => ListTile(
+            title: Text(communities[i].name ?? '@${communities[i].name}'),
```
Comment (shilangyu) [inline lib/pages/communities_list.dart:32]: The name is never `null`

File: lib/pages/communities_list.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+class CommunitiesListPage extends StatelessWidget {
+  final String title;
+  final List<CommunityView> communities;
+
+  const CommunitiesListPage({Key key, @required this.communities, this.title})
+      : assert(communities != null),
+        super(key: key);
+
+  void goToUser(BuildContext context, int id) {
```
Comment (shilangyu) [inline lib/pages/communities_list.dart:15]: should be go to community

File: lib/pages/instance.dart [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+            ),
+            ...site.banned.reversed.take(5).map((e) => ListTile(
+                  // subtitle: Text(),
+                  title: Text(
+                      (e.preferredUsername == null || e.preferredUsername == '')
+                          ? '@${e.name}'
+                          : e.preferredUsername),
+                  onTap: () => goToUser(e.id),
+                  leading: SizedBox(width: 50),
+                )),
```
Comment (shilangyu) [inline lib/pages/instance.dart:389]: Maybe we shouldn't show recently banned and just leave a button "show banned users"

File: lib/pages/instance.dart [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+          headerSliverBuilder: (context, innerBoxIsScrolled) => <Widget>[
+            SliverAppBar(
+              expandedHeight: 200,
+              floating: false,
+              pinned: true,
+              elevation: 0,
+              backgroundColor: theme.cardColor,
+              iconTheme: theme.iconTheme,
+              title: Text(
+                '${site.site.name}',
```
Comment (shilangyu) [inline lib/pages/instance.dart:146]: No need to interpolate a string into a string

File: lib/pages/instance.dart [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+            communities: commSnap.data,
+            title: 'Communities of ${site.site.name}'),
+      ));
+    }
+
+    return SingleChildScrollView(
+      child: SafeArea(
+        top: false,
+        child: Column(
+          // padding: EdgeInsets.only(top: 0),
```
Comment (shilangyu) [inline lib/pages/instance.dart:280]: remove if not used

File: lib/pages/instance.dart [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+                child: Text(
+                  'Recently banned:',
+                  style: theme.textTheme.headline6.copyWith(fontSize: 18),
+                ),
+              ),
+            ),
+            ...site.banned.reversed.take(5).map((e) => ListTile(
+                  // subtitle: Text(),
+                  title: Text(
+                      (e.preferredUsername == null || e.preferredUsername == '')
```
Comment (shilangyu) [inline lib/pages/instance.dart:384]: use `.isEmpty`

File: lib/pages/instance.dart [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+            ListTile(
+              title: Center(
+                child: Text(
+                  'Recently banned:',
+                  style: theme.textTheme.headline6.copyWith(fontSize: 18),
+                ),
+              ),
+            ),
+            ...site.banned.reversed.take(5).map((e) => ListTile(
+                  // subtitle: Text(),
```
Comment (shilangyu) [inline lib/pages/instance.dart:382]: remove unused

**Review by shilangyu — CHANGES_REQUESTED** [review]

**Review by shilangyu — APPROVED** [review]

**Pull request merged**